### PR TITLE
GCP-21 Add get_resources_type and look for it in get_relationships

### PR DESCRIPTION
**Pull request by funkyHat**
JIRA: GCP-21

### CODE REVIEW

File: cloudify_gcp/utils.py [hunk 1]
```diff
@@ -420,17 +421,27 @@ def get_relationships(
...
         filter_nodes = [filter_nodes]
     results = []
     for rel in relationships:
-        if filter_relationships and rel.type not in filter_relationships:
-            rel = None
-        if filter_nodes and rel.target.node.type not in filter_nodes:
-            rel = None
-        if rel:
+        res_type = get_resource_type(rel.target)
+        if not any([
```
Comment (idanmo) [inline cloudify_gcp/utils.py]: Please add a comment about what this if is actually checking...

File: cloudify_gcp/utils.py [hunk 2]
```diff
@@ -478,3 +489,7 @@ def get_net_and_subnet(ctx):
 
 def get_network(ctx):
     return get_net_and_subnet(ctx)[0]
+
+
+def get_resource_type(ctx):
```
Comment (idanmo) [inline cloudify_gcp/utils.py]: What does kind = False mean? why not None?

File: cloudify_gcp/utils.py (hunk 1 above)
Comment (funkyHat) [inline cloudify_gcp/utils.py]: Addressed below

File: cloudify_gcp/utils.py (hunk 2 above)
Comment (funkyHat) [inline cloudify_gcp/utils.py]: Fixed

**Pull request merged**